src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,512 @@
+| `stagehand`     |  `dart create`                                           | [2.14](https://github.com/dart-lang/stagehand/issues/671)        | 2.14*  |
+| `dartfmt`       | [`dart format`](https://dart.dev/tools/dart-format)      | [2.14](https://github.com/dart-lang/dart_style/issues/986)        | [2.15](https://github.com/dart-lang/dart_style/issues/986)            |
+| `dart2native`   | [`dart compile exe`](https://dart.dev/tools/dart-compile)    | [2.14](https://github.com/dart-lang/sdk/commit/cac00e9d956a6f7ef28628989912d971f6b908d4)        | [2.15](https://github.com/dart-lang/sdk/commit/6c5fb84716b1f257b170351efe8096fe2af2809b)            |
+| `dart2js`       | [`dart compile js`](https://dart.dev/tools/dart-compile)    | [2.17](https://github.com/dart-lang/sdk/commit/8415b70e75b1d5bbe8251fa6a9eab2d970cf9eec)         | [2.18](https://github.com/dart-lang/sdk/commit/69249df50bcc7a0489176efd3fd79fff018f1b91)             |
+| `dartdevc`      | [`webdev`](/tools/webdev)                              | [2.17](https://github.com/dart-lang/sdk/commit/5173fd2d224f669fd8d0a1d21adbfd6187d10f53)         | [2.18](https://github.com/dart-lang/sdk/commit/69249df50bcc7a0489176efd3fd79fff018f1b91)             |
+| `dartanalyzer`  | [`dart analyze`](https://dart.dev/tools/dart-analyze)    | [2.16](https://github.com/dart-lang/sdk/commit/f7af5c5256ee6f3a167f380722b96e8af4360b46)         | [2.18](https://github.com/dart-lang/sdk/issues/48457)             |
+| `dartdoc`       | `dart doc`                                                      | [2.16](https://github.com/dart-lang/sdk/issues/44610)         | [2.17](https://dart-review.googlesource.com/c/sdk/+/228647)             |
+| `pub`           | [`dart pub`](https://dart.dev/tools/dart-pub)            | [2.15](https://github.com/dart-lang/pub/issues/2736)         | [2.17](https://dart-review.googlesource.com/c/sdk/+/234283)             |
+{:.table .table-striped .nowrap}
+
+### Null safety migration tools
+
+  * The following null safety migration commands have been removed,
+    as Dart 3 [doesn't support code without null safety](#100-sound-null-safety):
+    - `dart migrate`
+    - `dart pub upgrade --null-safety`
+    - `dart pub outdated --mode=null-safety`
+
+#### Scope
+
+This is an *unversioned* change, that applies to all Dart 3 code.
+
+#### Symptom
+
+These commands will fail.
+
+#### Migration
+
+Use Dart 2.19 to [migrate to null safety](/null-safety/migration-guide).
+
+### Analyzer config
+
+The [analyzer configuration options](/guides/language/analysis-options#enabling-additional-type-checks)
+for enabling stricter checking have changed.
+
+#### Scope
+
+This is an *unversioned* change, that applies to all Dart 3 code.
+
+#### Symptom
+
+The former configuration options will fail with a warning like:
+
+```
+The option 'implicit-casts' is no longer supported.
+Try using the new 'strict-casts' option.
+```
+
+#### Migration
+
+Replace the analyzer config:
```

Comment:
Is this a file? Should we link to this page (if this is the same thing): https://dart.dev/guides/language/analysis-options#the-analysis-options-file

Reply:
This is already linked like 10 lines further up